### PR TITLE
Move accounts to use new connect client

### PR DESCRIPTION
**Pull request by marcserrat**
Switched accounts action to new Connect (openapi) Client

### CODE REVIEW

File: cnctcli/actions/products/export.py [hunk 1]
```diff
@@ -101,15 +105,14 @@ def _fill_item_row(ws, row_idx, item):
...
+def _dump_items(ws, client, product_id, silent):
     _setup_items_header(ws)
 
     processed_items = 0
     row_idx = 2
-    limit = 2
-    offset = 0
 
-    count, items = get_items(api_url, api_key, product_id, limit, offset)
+    items = client.products[product_id].items.filter()
```
Comment (ffaraoneim) [inline cnctcli/actions/products/export.py:114]: .all() better than .filter()

File: cnctcli/api/products.py [hunk 2]
```diff
@@ -3,164 +3,91 @@
...
-    )
-    if res.status_code != 204:
-        handle_http_error(res)
+def delete_item(client, product_id, item_id):
+    try:
+        (
+            client.products[product_id]
+            .items
+            .delete(item_id)
+        )
```
Comment (ffaraoneim) [inline cnctcli/api/products.py:90]: this doesn't work.

Should be: client.products[product_id].items[item_id].delete()

**Review by ffaraoneim — CHANGES_REQUESTED** [review]

File: cnctcli/actions/accounts.py [hunk 3]
```diff
@@ -4,32 +4,32 @@
...
-        account_id = account_data['id']
-        name = account_data['name']
+
+    try:
+        client = ConnectClient(
+            api_key=api_key,
+            endpoint=endpoint
+        )
+        account_data = client.accounts.filter().first()
+        breakpoint()
```
[the remark this answers is not in the document]
Comment (marcserrat) [reply cnctcli/actions/accounts.py]: comment from outdated version ;)

File: cnctcli/actions/products/export.py [hunk 4]
```diff
@@ -101,15 +105,14 @@ def _fill_item_row(ws, row_idx, item):
...
+def _dump_items(ws, client, product_id, silent):
     _setup_items_header(ws)
 
     processed_items = 0
     row_idx = 2
-    limit = 2
-    offset = 0
 
-    count, items = get_items(api_url, api_key, product_id, limit, offset)
+    items = client.products.resource(product_id).items.filter()
```
[the remark this answers is not in the document]
Comment (marcserrat) [reply cnctcli/actions/products/export.py]: correct

File: cnctcli/api/products.py [hunk 5]
```diff
@@ -3,164 +3,87 @@
...
-        f'{endpoint}/products/{product_id}/items/{item_id}',
-        headers=get_headers(api_key),
-    )
-    if res.status_code != 204:
-        handle_http_error(res)
+def delete_item(client, product_id, item_id):
+    try:
+        client.products[product_id].items[item_id].delete()
+    except ClientError as error:
+        if error.status_code != 204:
```
Comment (ffaraoneim) [inline cnctcli/api/products.py:88]: This is always true, errors are from status >= 400

**Review by ffaraoneim — CHANGES_REQUESTED** [review]

File: cnctcli/api/products.py (hunk 5 above)
Comment (marcserrat) [reply cnctcli/api/products.py:88]: awesome, did not saw...then totally unneded check

**Review by ffaraoneim — APPROVED** [review]

**Pull request merged**